TNone.value typed as undefined breaks the compilation

Given a piece of Typescript code that imports anything defined in disjunctions.d.ts like the following
```typescript
import { Right } from 'funfix';

Right('foo');
```
I get this error:
```
error TS2416: Property 'value' in type 'TNone' is not assignable to the same property in base type 'Option<never>'.
  Type 'undefined' is not assignable to type 'never'.
```
This is because `TNone.value` has type `undefined` while it should be `never` as `TNone` extends `Option<never>`. Somehow the tsc has troubles dealing with the union type defined by `Option<A>.value` which is `undefined | A`.

This problem can be fixed by updating the `value` definition to 
```typescript
public readonly value!: never
```

I am available to raise a PR if this is the correct way to go.
I am seeing this issue with `funfix 7.0.1`, compiling with `tsc 2.7.2`.

--- a/AUTHORS
+++ b/AUTHORS
@@ -7,3 +7,5 @@ Oliver Joseph Ash
 https://github.com/OliverJAsh
 Alvaro Carrasco
 https://github.com/alvaroc1
+Matteo Castellini
+https://github.com/mcastellini

--- a/packages/funfix-core/src/disjunctions.ts
+++ b/packages/funfix-core/src/disjunctions.ts
@@ -1106,7 +1106,7 @@ export function Some<A>(value: A): TSome<A> {
  * non-empty values in the [[Option]] disjunction.
  */
 export class TNone extends Option<never> {
-  public readonly value!: undefined
+  public readonly value!: never
   private constructor() { super(undefined, true) }
 }
 

